src/main/java/org/isf/admission/gui/AdmissionBrowser.java

```diff
@@ -1741,422 +1700,376 @@ private JButton getSaveButton() {
-    	DateFormat currentDateFormat = DateFormat.getDateInstance(DateFormat.SHORT, new Locale(GeneralData.LANGUAGE));
-    	List<OHExceptionMessage> errors = new ArrayList<>();
-		Date beginDate,endDate;
-		if (admission.getAdmDate()!=null)beginDate=admission.getAdmDate().getTime();else beginDate=null;
-		if (admission.getDisDate()!=null)endDate=admission.getDisDate().getTime();else endDate=null;
-		for (org.isf.operation.model.OperationRow opRow : list) {
-			Date currentRowDate = opRow.getOpDate().getTime();
-			/*
-			 * prevent for fails due to time 
-			 */
-			currentRowDate.setHours(23);
-			currentRowDate.setMinutes(59);
-			currentRowDate.setSeconds(59);
-			
-			if ((beginDate!=null)&&(endDate!=null)){
-				if ((currentRowDate.before(beginDate))||(currentRowDate.after(endDate))){
-					errors.add(new OHExceptionMessage(MessageBundle.getMessage("angal.hospital"), 
-							MessageBundle.getMessage("angal.admission.invalidoperationdate") + " " 
-							+ MessageBundle.getMessage("angal.admission.theoperationdatebetween") + " "
-                            + currentDateFormat.format(beginDate) + " " + MessageBundle.getMessage("angal.admission.and") + " " 
-							+ currentDateFormat.format(endDate), OHSeverityLevel.ERROR));
+		String query = s.trim();
+		ArrayList<Disease> results = new ArrayList<>();
+		for (Disease disease : diseaseList) {
+			if (!query.equals("")) {
+				String[] patterns = query.split(" ");
+				String name = disease.getDescription().toLowerCase();
+				boolean patternFound = false;
+				for (String pattern : patterns) {
+					if (name.contains(pattern.toLowerCase())) {
+						patternFound = true;
+						//It is sufficient that only one pattern matches the query
+						break;
+					}
 				}
-			}
-			if ((beginDate!=null)&&(endDate==null)){
-				if (currentRowDate.before(beginDate)){
-					errors.add(new OHExceptionMessage(MessageBundle.getMessage("angal.hospital"), 
-							MessageBundle.getMessage("angal.admission.invalidoperationdate") + " " 
-							+ MessageBundle.getMessage("angal.admission.theoperationdatenewerthan") + " "
-                            + currentDateFormat.format(beginDate), OHSeverityLevel.ERROR));
+				if (patternFound) {
+					results.add(disease);
 				}
+			} else {
+				results.add(disease);
 			}
 		}
-		return errors;
+		return results;
 	}
+        
+    @SuppressWarnings("deprecation")
+    public List<OHExceptionMessage> checkAllOperationRowDate(List<OperationRow> list, Admission admission) {
+	    DateFormat currentDateFormat = DateFormat.getDateInstance(DateFormat.SHORT, new Locale(GeneralData.LANGUAGE));
+	    List<OHExceptionMessage> errors = new ArrayList<>();
+	    Date beginDate;
+	    Date endDate;
+	    if (admission.getAdmDate() != null)
+		    beginDate = admission.getAdmDate().getTime();
+	    else
+		    beginDate = null;
+	    if (admission.getDisDate() != null)
+		    endDate = admission.getDisDate().getTime();
+	    else
+		    endDate = null;
```

Comment:
code block missing?

Reply:
Yeah; the reformatting doesn't automatically add them; I will do that the next time through the file.

Reply:
It was easy to do now so I did it.

Reply:
Alright, thanks!